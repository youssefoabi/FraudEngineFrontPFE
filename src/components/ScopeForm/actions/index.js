export const SAVE_CREATE_SCOPE_REQUEST = 'SAVE_CREATE_SCOPE_REQUEST';
export const SAVE_CREATE_SCOPE_SUCCESS = 'SAVE_CREATE_SCOPE_SUCCESS';
export const SAVE_CREATE_SCOPE_ERROR = 'SAVE_CREATE_SCOPE_ERROR';
export const SAVE_EDIT_SCOPE_REQUEST = 'SAVE_EDIT_SCOPE_REQUEST';
export const SAVE_EDIT_SCOPE_SUCCESS = 'SAVE_EDIT_SCOPE_SUCCESS';
export const SAVE_EDIT_SCOPE_ERROR = 'SAVE_EDIT_SCOPE_ERROR';
export const RESET_SCOPE_SAVE_ERROR = 'RESET_SCOPE_SAVE_ERROR';
export const RESET_SCOPE_SAVE_SUCCESS = 'RESET_SCOPE_SAVE_SUCCESS';
