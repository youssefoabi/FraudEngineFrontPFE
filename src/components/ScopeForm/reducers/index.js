import { navigationScopeError, navigationScopeSuccess } from './navigationReducer';

export { navigationScopeError, navigationScopeSuccess };
