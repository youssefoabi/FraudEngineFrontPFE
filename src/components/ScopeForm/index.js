import { func } from 'prop-types';
import { Grid, Snackbar } from '@material-ui/core';
import { path } from 'ramda';
import { useDispatch, useSelector } from 'react-redux';
import MuiAlert from '@material-ui/lab/Alert';
import React, { useCallback } from 'react';

import { RESET_SCOPE_SAVE_ERROR, RESET_SCOPE_SAVE_SUCCESS } from './actions';
import ScopeDataEdit, { scopeShape } from '../ScopeDataEdit';
import ScopeEditionHelper from '../ScopeEditionHelper';

const ScopeForm = ({ scope, setScope }) => {
  const onDataEditChange = ({ id, value }) =>
    setScope(prevScope => ({
      ...prevScope,
      [id]: value,
    }));

  const dispatch = useDispatch();

  const onCloseErrorSnackbar = useCallback(() => dispatch({ type: RESET_SCOPE_SAVE_ERROR }), [dispatch]);
  const onShowSuccessSnackbar = useCallback(() => dispatch({ type: RESET_SCOPE_SAVE_SUCCESS }), [dispatch]);

  const error = useSelector(path(['navigation', 'scopeFormError']));
  const success = useSelector(path(['navigation', 'scopeFormSuccess']));

  return (
    <Grid container direction="column">
      <ScopeDataEdit scope={scope} handleChange={onDataEditChange} />
      <ScopeEditionHelper scope={scope} />
      {error?.errorMessage && (
        <Snackbar open autoHideDuration={6000} onClose={onCloseErrorSnackbar}>
          <MuiAlert onClose={onCloseErrorSnackbar} severity="error">
            {error.errorMessage}
          </MuiAlert>
        </Snackbar>
      )}
      {success?.scopeSaveSuccess && (
        <Snackbar open autoHideDuration={6000} onClose={onShowSuccessSnackbar}>
          <MuiAlert onClose={onShowSuccessSnackbar} severity="success">
            La gamme a bien été enregistrée
          </MuiAlert>
        </Snackbar>
      )}
    </Grid>
  );
};

ScopeForm.propTypes = {
  scope: scopeShape,
  setScope: func,
};

ScopeForm.defaultProps = {
  scope: null,
  setScope: Function.prototype,
};

export default ScopeForm;
