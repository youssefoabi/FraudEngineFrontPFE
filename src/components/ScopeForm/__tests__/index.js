import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import ScopeForm from '../';

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
  useSelector: jest
    .fn()
    .mockReturnValueOnce({
      errorMessage: 'ERROR',
    })
    .mockReturnValueOnce({
      scopeSaveSuccess: false,
    })
    .mockReturnValueOnce({})
    .mockReturnValueOnce({
      scopeSaveSuccess: true,
    }),
}));

describe(ScopeForm.name, () => {
  describe('with error', () => {
    itRendersCorrectly(<ScopeForm />);
  });
  describe('with success', () => {
    itRendersCorrectly(<ScopeForm />);
  });
  describe('without Snackbar', () => {
    itRendersCorrectly(<ScopeForm />);
  });
});
