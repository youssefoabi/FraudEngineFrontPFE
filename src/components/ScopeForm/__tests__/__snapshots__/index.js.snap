// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopeForm with error renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <ScopeDataEdit
    handleChange={[Function]}
    scope={null}
  />
  <ScopeEditionHelper
    scope={null}
  />
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={6000}
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(Alert))
      onClose={[Function]}
      severity="error"
    >
      ERROR
    </WithStyles(ForwardRef(Alert))>
  </WithStyles(ForwardRef(Snackbar))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`ScopeForm with success renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <ScopeDataEdit
    handleChange={[Function]}
    scope={null}
  />
  <ScopeEditionHelper
    scope={null}
  />
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={6000}
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(Alert))
      onClose={[Function]}
      severity="success"
    >
      La gamme a bien été enregistrée
    </WithStyles(ForwardRef(Alert))>
  </WithStyles(ForwardRef(Snackbar))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`ScopeForm without Snackbar renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <ScopeDataEdit
    handleChange={[Function]}
    scope={null}
  />
  <ScopeEditionHelper
    scope={null}
  />
</WithStyles(ForwardRef(Grid))>
`;
